### DISCUSSION TITLE
Kafka Storage Engine - Is Data Loss Possible ?

### DISCUSSION THREAD

**Discussion by karthik-thiyagarajan** [ClickHouse/ClickHouse#52298] [Q&A] [answered]
We have a ClickHouse Table (clickhouse/clickhouse:23.3 version) created with Engine=Kafka with 2 consumers. When there is a high flow of inbound messages, we see the following two errors : 

(i) "Too many parts (600). Merges are processing significantly slower than inserts"
(ii) DB::Exception: Too many partitions for single INSERT block (more than 100)

Here is my understanding : I could not find any documentation that says that these kind of exceptions will lead to a data loss or not. However, based on my limited reading, if it comes through Kafka route and if the messages are coming with "Engine=Kafka", and if there are any exceptions while trying to get them added to Table, then it is my understanding that these messages wont be committed in Kafka and hence those uncommitted messages will be kept in loop for back off and retry until the Kafka Retention Period is active. It will lead to data loss after the messages are deleted in Kafka (after retention period). 

Can someone confirm my understanding.

**Comment by den-crane** [comment, marked as answer]
>It will lead to data loss after the messages are deleted in Kafka (after retention period).

Yes. But if you define kafka  retention period to 1 second they also will be lost. How it's related to Clickhouse server?

**Reply by karthik-thiyagarajan** [reply]
Since I was talking about ClickHouse Table created with Kafka Engine, I wanted to confirm the behaviour. Thanks.
